pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -414,9 +414,7 @@ func (r *RepoSyncReconciler) mapSecretToRepoSyncs(secret client.Object) []reconc
 			// It is a one-to-one mapping between the copied ns-reconciler Secret and the RepoSync object,
 			// so requeue the mapped RepoSync object and then return.
 			reconcilerName := core.NsReconcilerName(rs.GetNamespace(), rs.GetName())
-			isGitSecret := rs.Spec.SourceType == string(v1beta1.GitSource) && rs.Spec.Git != nil &&
-				secret.GetName() == ReconcilerResourceName(reconcilerName, rs.Spec.SecretRef.Name)
-			if isGitSecret {
+			if isUpsertedSecret(&rs, secret.GetName()) {
```

Comment:
The change only triggers the reconciliation when the upserted secret is updated.

We need to requeue the RepoSync objects when the user secret is updated as well. That requires us to find all attached RepoSync objects: https://github.com/GoogleContainerTools/kpt-config-sync/pull/12/commits/199d96e3767d37b855d28db4792485f98d881067#diff-823453b6c89ee795b8bca312adb26bf8b3e28e3cec8b89a8af6edf028548a6cbR441

Reply:
Good catch, thank you! I've updated the change accordingly and added an e2e test, PTAL